controllers/keystoneapi_controller.go

```diff
@@ -880,6 +880,21 @@ func (r *KeystoneAPIReconciler) reconcileNormal(
 	// Create ConfigMaps and Secrets required as input for the Service and calculate an overall hash of hashes
 	//
 
+	// ensure MariaDBAccount exists.  This account record may be created by
+	// openstack-operator or the cloud operator up front without a specific
+	// MariaDBDatabase configured yet.   Otherwise, a MariaDBAccount CR is
+	// created here with a generated username as well as a secret with
+	// generated password.   The MariaDBAccount is created without being
+	// yet associated with any MariaDBDatabase.
+	_, _, err = mariadbv1.EnsureMariaDBAccount(
+		ctx, helper, instance.Spec.DatabaseAccount,
+		instance.Namespace, false,
+	)
+
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
it might be good to have a condition which reflects the status of the MariaDBAccount. if not it would allow to see where it is stuck in reconcile.

Reply:
OK, I would add this to all the operators I'm doing...

Reply:
hmm glance-operator does the conditions differently, it already applies an error for this in its conditions, just not a specific one.    guess this will have to be case by case